### ISSUE TITLE
Reject changes reverts properties on complex classes to (very) old values.

### ISSUE THREAD

**Issue by H-a-w-k** [Breeze/breeze.js#171]
```
public class  A {
    public B myB {get; set;}
}
[ComplexType]
public class  B {
    public double myDouble {get; set;}
}

//in ef database
var myA = new A{
    myB = new B{
        myDouble = 1
    }
}
```
If I on the client changes myDouble to 3 and call rejectChanges, myDouble is 1 and all is good.
If I on the client changes myDouble to 3 and call saveChanges, myDouble is 3, it saves perfectly and all is good.
If  on the client changes myDouble to 5 and call rejectChanges,, myDouble is 1, even though it's 3 on the server. If I refresh the page, the "base number" to which it rejects is then 3, but if I alter it to 5, save, alter it to 7 and reject, I get 3.

Is this desired behavior?

**Comment by steveschmitt** [comment]
That sounds like a bug.  `rejectChanges` should revert to the last saved value.

**Comment by steveschmitt** [comment]
I have reproduced the problem, and a fix will be in the next release of Breeze.  Thanks for finding this.

**Comment by H-a-w-k** [comment]
No worries. We rely heavily on complex types so this is important to us :)

**Comment by steveschmitt** [comment]
Fixed in breeze 1.5.15
